### PR TITLE
chore: move Gateway API types and consts aliases to a separate package

### PR DESCRIPTION
**Pull request by czeslavo**
**What this PR does / why we need it**:

Moves Gateway API types and consts aliases to a separate package (`internal/util/gatewayapi`). Also, adds aliases that were missing so that we no longer have direct references to `sigs.k8s.io/gateway-api/apis/v1alpha2` or `sigs.k8s.io/gateway-api/apis/v1beta1` other than their respective `SchemeGroupVersion`, `GroupVersion` and `Install` in the rest of the codebase.

**Which issue this PR fixes**:

Closes https://github.com/Kong/kubernetes-ingress-controller/issues/3503.

### CODE REVIEW

**Comment by programmer04** [comment]
Is it possible to enforce it with a linter configuration?

**Review by programmer04 — COMMENTED** [review]
Is it possible to enforce it with a linter configuration?

File: internal/util/gatewayapi/aliases.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
+package gatewayapi
+
+import (
+	gatewayv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+// This file contains aliases for types and consts from the Gateway API.  Its purpose is to allow easily migrate from
```
Comment (pmalek) [inline internal/util/gatewayapi/aliases.go:8]:
```suggestion
// This file contains aliases for types and consts from the Gateway API.  Its purpose is to allow to easily migrate from
```

File: internal/util/gatewayapi/contraints.go
Comment (pmalek) [inline internal/util/gatewayapi/contraints.go:1]: Would it be difficult to move this to a package that's not under `util` (ditto for `aliases.go`)? I guess not.

How about `internal/gatewayapi` or `internal/types/gatewaypi`?

File: internal/util/gatewayapi/aliases.go (hunk 1 above)
Comment (pmalek) [reply internal/util/gatewayapi/aliases.go:8]: of `...to allow easy migration ...`

File: internal/util/gatewayapi/contraints.go
Comment (czeslavo) [reply internal/util/gatewayapi/contraints.go:1]: Moved to `internal/gatewayapi`.

**Review by pmalek — APPROVED** [review]

**Pull request merged**